c/tskit/tables.c

```
@@ -6977,71 +7038,62 @@ tsk_table_collection_load(
     if (ret != 0) {
         ret = tsk_set_kas_error(ret);
         goto out;
     }
-    ret = tsk_table_collection_read_format_data(self);
+    ret = tsk_table_collection_read_format_data(self, &store);
     if (ret != 0) {
         goto out;
     }
-    ret = tsk_node_table_load(&self->nodes, self->store);
+    ret = tsk_node_table_load(&self->nodes, &store);
     if (ret != 0) {
         goto out;
     }
-    ret = tsk_edge_table_load(&self->edges, self->store);
+    ret = tsk_edge_table_load(&self->edges, &store);
     if (ret != 0) {
         goto out;
     }
-    ret = tsk_site_table_load(&self->sites, self->store);
+    ret = tsk_site_table_load(&self->sites, &store);
     if (ret != 0) {
         goto out;
     }
-    ret = tsk_mutation_table_load(&self->mutations, self->store);
+    ret = tsk_mutation_table_load(&self->mutations, &store);
     if (ret != 0) {
         goto out;
     }
-    ret = tsk_migration_table_load(&self->migrations, self->store);
+    ret = tsk_migration_table_load(&self->migrations, &store);
     if (ret != 0) {
         goto out;
     }
-    ret = tsk_individual_table_load(&self->individuals, self->store);
+    ret = tsk_individual_table_load(&self->individuals, &store);
     if (ret != 0) {
         goto out;
     }
-    ret = tsk_population_table_load(&self->populations, self->store);
+    ret = tsk_population_table_load(&self->populations, &store);
     if (ret != 0) {
         goto out;
     }
-    ret = tsk_provenance_table_load(&self->provenances, self->store);
+    ret = tsk_provenance_table_load(&self->provenances, &store);
     if (ret != 0) {
         goto out;
     }
-    ret = tsk_table_collection_load_indexes(self);
+    ret = tsk_table_collection_load_indexes(self, &store);
     if (ret != 0) {
         goto out;
     }
-    ret = tsk_table_collection_check_offsets(self);
 out:
+    /* Any read errors will have already happened so we ignore any errors here. */
+    kastore_close(&store);
     return ret;
```

I don't quite get this - in the case where there have been no read errors we're ignoring errors from `kastore_close`, which can still return errors, even when the mode is `KAS_READ`.

---

Hmm, yes. You're right, I'll update.